Core/HttpHandlerProxy.cs

```diff
@@ -0,0 +1,151 @@
+            {
+                return base.SendAsync(request, cancellationToken);
+            }
+        }
+
+        private readonly string httpClientName;
+        private readonly HttpMessageHandler httpMessageHandler;
+        private readonly HttpClientHandler httpClientHandler;
+
+        /// <summary>
+        /// Gets or sets the cookie container.
+        /// </summary>
+        /// <value>The cookie container.</value>
+        public CookieContainer CookieContainer { get; set; }
+
+        /// <summary>
+        /// Gets or sets authentication information for the request.
+        /// </summary>
+        /// <returns>An <see cref="T:System.Net.ICredentials"/> that contains the authentication credentials associated with the request. The default is null.</returns>
+        public ICredentials Credentials { get; set; }
+
+        /// <summary>
+        /// Gets or sets proxy information for the request.
+        /// </summary>
+        public IWebProxy Proxy { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that indicates whether to send an authenticate header with the request.
+        /// </summary>
+        /// <returns>true to send a WWW-authenticate HTTP header with requests after authentication has taken place; otherwise, false. The default is false.</returns>
+        public bool PreAuthenticate { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="T:System.Boolean"/> value that controls whether default credentials are sent with requests.
+        /// </summary>
+        /// <returns>true if the default credentials are used; otherwise false. The default value is false.</returns>
+        public bool UseDefaultCredentials { get; set; }
+
+        public HttpHandlerProxy(IHttpMessageHandlerFactory httpMessageHandlerFactory, string httpClientName)
+        {
+            this.httpClientName = httpClientName;
+            this.httpMessageHandler = httpMessageHandlerFactory?.CreateHandler(httpClientName);
+            this.httpClientHandler = null == httpMessageHandlerFactory ? HttpClientExtensions.CreateHttpClientHandler(httpClientName) : null;
+        }
+
+        private void Authorize(HttpRequestMessage request)
+        {
+            switch (Credentials)
+            {
+                case NetworkCredential networkCredential:
+                    {
+                        string authString = !string.IsNullOrEmpty(networkCredential.Domain)
+                            ? networkCredential.Domain + "\\" + networkCredential.UserName + ":" + networkCredential.Password
+                            : networkCredential.UserName + ":" + networkCredential.Password;
+
+                        string base64AuthString = Convert.ToBase64String(Encoding.UTF8.GetBytes(authString));
+
+                        request.Headers.Authorization = new AuthenticationHeaderValue("Basic", base64AuthString);
+                    }
+                    break;
+
+                case null:

```

Comment:
1. It also may be https://learn.microsoft.com/en-us/dotnet/api/system.net.credentialcache?view=net-7.0 class
2. Formally both NetworkCredential and CredentialCache implement ICredentialsByHost  - don't know if we need to support it.

Reply:
for this case it is simpler to leave current logic, don't see any benefit of using https://learn.microsoft.com/en-us/dotnet/api/system.net.credentialcache?view=net-7.0

Reply:
depends on who will be using your library... at least this code will throw on unknown creds type so ok